tests/testthat/test-run.R

```diff
@@ -34,6 +34,31 @@ test_that("run_one", {
   wipe_results()
 })
 
+test_that("cases can be versioned", {
+  bm_unversioned <- Benchmark(
+    "unversioned",
+    setup = function(x = c('foo', 'bar')) { cat(x) }
+  )
+  res_unversioned <- run_benchmark(bm_unversioned)
+  lapply(res_unversioned$results, function(result) {
+    # when version is not supplied, it should not appear in tags
+    expect_false("case_version" %in% names(result$tags))
+  })
+
+  bm_versioned <- Benchmark(
+    "versioned",
+    setup = function(x = c('foo', 'bar')) cat(x),
+    case_version = function(params) c("foo" = 1L, "bar" = 2L)[params$x]
```

Comment:
Because of how I constructed this example, if `params` were `list(x = "baz")` (or any novel value), it will return `NA`, which will get written to the tags as `"tags": {..., "case_version": null}`, which I _think_ may break history in conbench (though I'm not 💯).